### PR TITLE
CO-211: quarantine repeated probe-timeout churn

### PR DESCRIPTION
**Pull request by Kbediako**
## What
- Retain local provider-intake diagnostics when the control-host `co-status` supervision probe times out.
- Quarantine repeated same-active-worker `probe_timeout` churn only after one fail-closed timeout restart, and only when retained polling truth is `provider_refresh_lifecycle_stuck` with `restart_required=true`.
- Refresh the CO-211 Rework docs packet and add focused regression coverage for retained diagnostics, timeout quarantine, non-lifecycle timeout fail-closed behavior, and CO-194 stale terminal claims.

## Why
CO-211 was reopened after Apr 21 evidence showed repeated `probe_timeout` / restart-required churn while active provider workers continued heartbeating. The prior restart-required quarantine did not cover timeout restart records with missing diagnostics, so future agents could not reconstruct the same active-worker series cleanly.

## Validation
- `npm run test:orchestrator -- orchestrator/tests/ControlHostSupervision.test.ts` passed: 74 tests.
- `node scripts/delegation-guard.mjs` passed.
- `node scripts/spec-guard.mjs --dry-run` passed.
- `npm run build` passed.
- `npm run lint` passed with existing `no-explicit-any` warnings in `orchestrator/tests/DelegationMcpHealth.test.ts`.
- `npm run test` passed: 346 files, 4436 tests.
- `git diff --check` passed.
- `npm run docs:check` passed.
- `npm run docs:freshness` passed.
- `npm run repo:stewardship` passed.
- `node scripts/diff-budget.mjs` passed: 10 files / 632 lines.
- `PACK_SMOKE_ALLOW_MARKETPLACE_SKIP=1 PACK_SMOKE_MARKETPLACE_SKIP_REASON="local Codex CLI lacks marketplace add support in this worker environment; package CLI/review/skills smoke still runs" npm run pack:smoke` passed.

## Review Notes
Manifest-backed standalone review was attempted under `FORCE_CODEX_REVIEW=1`, but the review process stalled/drifted without a concrete verdict. I performed a manual correctness review and fixed the one finding: timeout quarantine now requires retained lifecycle-stuck `restart_required` truth, with a non-lifecycle timeout regression added. An explicit elegance/minimality pass kept the diff scoped to supervision evaluation, timeout diagnostic fallback, focused tests, and the CO-211 docs mirrors.

Post-recovery status caveat: direct worker-local `co-status --format json` was blocked by EPERM/stale endpoint in this environment. Shared-root provider-intake state at `2026-04-21T11:40:10.645Z` showed `polling.stuck=false`, `polling.restart_required=false`, and running workers `CO-211`, `CO-225`, and `CO-268`; this is recorded in the Linear workpad rather than overclaiming direct command success.

## Summary by CodeRabbit

* **New Features**
  * Captures timeout diagnostics from persisted local provider state so timeouts include diagnostic context
  * Quarantines repeated same-worker timeout restarts to prevent churn while active workers remain healthy
  * Distinguishes first/unrelated timeouts from repeated same-worker probe-timeout sequences

* **Documentation**
  * Updated specs, PRD, action plan and task tracking to reflect timeout recurrence handling and diagnostic-retention rules

* **Tests**
  * Added tests covering timeout diagnostics, quarantine behaviour and edge cases

### CODE REVIEW

File: orchestrator/src/cli/controlHostSupervisionCliShell.ts [hunk 1]
```diff
@@ -792,6 +813,72 @@ async function probeControlHostHealth(
...
+
+function resolveControlHostSupervisionProviderIntakeStatePath(
+  config: ControlHostSupervisionConfig,
+  env: NodeJS.ProcessEnv
+): string {
+  const configuredRunsDir = env.CODEX_ORCHESTRATOR_RUNS_DIR?.trim();
+  const runsRoot =
+    configuredRunsDir && configuredRunsDir.length > 0
+      ? resolve(config.repoRoot, configuredRunsDir)
+      : join(config.repoRoot, '.runs');
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply orchestrator/src/cli/controlHostSupervisionCliShell.ts]: Fixed in 2f4dfb99c: timeout diagnostic lookup now resolves from the effective `CODEX_ORCHESTRATOR_ROOT` that `co-status` receives, including relative `CODEX_ORCHESTRATOR_RUNS_DIR` handling. Added a focused regression covering env-root fallback diagnostics.

File: orchestrator/src/cli/control/controlHostSupervision.ts [hunk 2]
```diff
@@ -566,6 +567,79 @@ function resolveRepeatedActiveWorkerRestartQuarantine(input: {
...
+  polling: ControlHostSupervisionPollingDiagnostic | null
+): boolean {
+  if (!polling || polling.restart_required !== true) {
+    return false;
+  }
+  return (
+    polling.reason === 'provider_refresh_lifecycle_stuck' ||
+    polling.last_error === 'provider_refresh_lifecycle_stuck'
+  );
+}
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply orchestrator/src/cli/control/controlHostSupervision.ts:657]: @coderabbitai Fixed in bdb8cc7a8: timeout quarantine now receives `minPollingUpdatedAt` and refuses to quarantine retained polling snapshots older than the supervised child start. Added a focused regression proving stale pre-start provider-intake state stays fail-closed as `probe_timeout` instead of entering the quarantine path.
